rules/link-users-by-email-with-metadata.md

```diff
@@ -27,51 +27,49 @@ function (user, context, callback) {
    },
    qs: {
      search_engine: 'v2',
-     q: 'email:"' + user.email + '" -user_id:"' + user.user_id + '"',
+     q: 'email.raw:"' + user.email + '" AND email_verified: "true" -user_id:"' + user.user_id + '"',
    }
   },
   function(err, response, body) {
     if (err) return callback(err);
     if (response.statusCode !== 200) return callback(new Error(body));
 
     var data = JSON.parse(body);
-    if (data.length > 0) {
-      async.each(data, function(targetUser, cb) {
-        if (targetUser.email_verified && targetUser.email.toLowerCase() === user.email.toLowerCase()) {
-          var aryTmp = targetUser.user_id.split('|');
-          var provider = aryTmp[0];
-          var targetUserId = aryTmp[1];
+    if (data.length > 1) {
+      return callback(new Error('[!] Rule: Multiple user profiles already exist - cannot select base profile to link with'));
+    }
+    if (data.length === 0) {
+      console.log('[-] Skipping link rule');
+      return callback(null, user, context);
+    }
+    
+    var originalUser = data[0];
+    var aryTmp = user.user_id.split('|');
+    var provider = aryTmp[0];
+    var newUserId = aryTmp[1];
 
-          targetUser.app_metadata = targetUser.app_metadata || {};
-          targetUser.user_metadata = targetUser.user_metadata || {};
-          auth0.users.updateAppMetadata(user.user_id, targetUser.app_metadata)
-          .then(auth0.users.updateUserMetadata(user.user_id, targetUser.user_metadata))
-          .then(function(){
-            request.post({
-              url: userApiUrl + '/' + user.user_id + '/identities',
-              headers: {
-                Authorization: 'Bearer ' + auth0.accessToken
-              },
-              json: { provider: provider, user_id: targetUserId }
-            }, function(err, response, body) {
-                if (response && response.statusCode >= 400) {
-                  return cb(new Error('Error linking account: ' + response.statusMessage));
-                }
-                cb(err);
-            });
-          })
-          .catch(function(err){
-            cb(err);
-          });
-        } else {
-          cb();
-        }
-      }, function(err) {
-        callback(err, user, context);
+    user.app_metadata = user.app_metadata || {};
+    user.user_metadata = user.user_metadata || {};
+    auth0.users.updateAppMetadata(user.user_id, user.app_metadata)
```

Comment:
This doesn't look right. `user` is the user that's logging in now. `originalUser` is the existing user that we want to merge with `user`, keeping `originalUser`'s profile as the primary and making `user` the secondary.

As such, the current profile's metadata should be merged with `originalUser`'s metadata. [Per the SDK](https://github.com/auth0/node-auth0/tree/v2.0.0-alpha#users-metadata), I think `user.user_id` should be changed to `originalUser.user_id` and the SDK will take care of the merging by overwriting / updating. (It deletes only when `null` is set, AFAICT.)